src/main.cpp

```diff
@@ -3915,6 +3916,11 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
         }
         if (ShutdownRequested())
             return true;
+        nTxCheckLimitCurrent+=block.vtx.size();
+        if (nTxCheckLimit > 0 && nTxCheckLimitCurrent >= nTxCheckLimit && chainActive.Height() - pindex->nHeight >= std::min(nCheckDepth, MIN_BLOCK_TO_RESPECT_WITH_MAX_TX_CHECK)) {
```

Comment:
The std::min isn't necessary, as `chainActive.Height() - pindex->nHeight` will never exceed nCheckDepth (we bail out of the loop before that can happen).